### ISSUE TITLE
End of life of Solidity 0.4

### ISSUE THREAD

**Issue by alcuadrado** [ethereum/solidity#5654]
## Description

I couldn't find any information about Solidity 0.4 future plans. I'm specially interested about its EOL. I think it's really important to plan libraries' and tooling's migration to 0.5. Has it been defined?

**Comment by leonardoalt** [comment]
Hi @alcuadrado , as far as I know 0.4 will not be maintained at all. The idea has always been that everyone moves to 0.5, that's why many changes were available even before the release.

**Comment by alcuadrado** [comment]
Thanks for the quick answer @leonardoalt! 

Shouldn't this be clearly stated in 0.4 docs? There's this thing

![image](https://user-images.githubusercontent.com/176499/49946062-ee52fe00-fecc-11e8-8772-62d56a4c24b3.png)

but IMO it looks like a generic readthedocs message.

**Comment by leonardoalt** [comment]
No worries @alcuadrado . That message is probably just from readthedocs.
Why should it explicitly say it? Older versions haven't been kept maintained before, why would 0.4?

**Comment by alcuadrado** [comment]
I think there are a few reasons for users to assume that 0.4 would be maintained for some time:

1. Most mainstream language (a) avoid breaking changes (eg: javascript), or (b) maintain two versions of the language until the EOL of the older one (eg: python, ruby, java). IMO this leads to people expecting one of these release plans unless otherwise stated.

2. The bast majority of code is written in Solidity 0.4.  Until this situation changes, newcomers would be naturally inclined to use that version, and migrating to 0.5 is harder. 

3. Some of the most popular devtools for Solidity only have beta versions with support for Solidity 0.5 (eg: Truffle). This also complicates migrating to 0.5. 

I understand that maintaining two versions of a language is a lot of work and I'm OK with 0.4 not maintained. I just think that not everyone is aware of this. 

Also, not having clear information about this makes planning the migration to 0.5 harder. For example, I opened this issue as a result of a conversation with the Open Zeppelin team about their migration plans. During the conversation we realized that we had no idea about the EOL of 0.4. Without this info it's not clear if the Solidity 0.4 code should be maintained at all.
